backend/src/apiserver/server/visualization_server.go

```diff
@@ -78,11 +79,8 @@ func (s *VisualizationServer) validateCreateVisualizationRequest(request *go_cli
 // It returns the generated HTML as a string and any error that is encountered.
 func (s *VisualizationServer) generateVisualizationFromRequest(request *go_client.CreateVisualizationRequest) ([]byte, error) {
 	serviceURL := s.getVisualizationServiceURL(request)
-	if !isVisualizationServiceAlive(serviceURL) {
-		return nil, util.NewInternalServerError(
-			fmt.Errorf("service not available"),
-			"Service not available",
-		)
+	if err := isVisualizationServiceAlive(serviceURL); err != nil {
+		return nil, util.Wrap(err, "Unable to verify visualization service aliveness.")
```

Comment:
nit: `return nil, util.Wrap(err, "Cannot generate visualization.")`

Because the internal error message is already telling it failed to verify service is alive.

Reply:
Fixed.